sdk/state/state_test.go

```diff
@@ -255,29 +254,18 @@ func Test(t *testing.T) {
 
 		//// Sender: creates new Payment, sends to other party
 		sendingChannel.SetIterationNumber(i)
-		payment, err := sendingChannel.ProposePayment(amount)
-		require.NoError(t, err)
-		j, err := json.Marshal(payment)
+		payment, err := sendingChannel.ProposePayment(state.Amount{Asset: state.NativeAsset{}, Amount: amount})
 		require.NoError(t, err)
 
 		ci, di, err := sendingChannel.PaymentTxs(payment)
 		require.NoError(t, err)
 
 		//// Receiver: receives new payment proposal, validates, then confirms by signing both
 		receivingChannel.SetIterationNumber(i)
-		payment = &state.Payment{}
-		err = json.Unmarshal(j, payment)
-		require.NoError(t, err)
-
 		payment, err = receivingChannel.ConfirmPayment(payment)
 		require.NoError(t, err)
```

Comment:
This is not really specific to this PR, but this no error line seems suspicious. I think it should return an error since the payment isn't confirmed, because the sending channel hasn't signed the declaration yet.

Reply:
true. Going to leave this change as part of this ticket if that sounds good to you
https://github.com/stellar/experimental-payment-channels/issues/65